internal/pkg/aws/acm/acm.go

```diff
@@ -42,14 +43,15 @@ func New(s *session.Session) *ACM {
 // ValidateCertAliases validates if aliases are all valid against the provided ACM certificates.
 func (a *ACM) ValidateCertAliases(aliases []string, certs []string) error {
 	validAliases := make(map[string]bool)
+	AllowedDomainsCert := make(map[string][]string)
 	ctx, cancelWait := context.WithTimeout(context.Background(), waitForFindValidAliasesTimeout)
 	defer cancelWait()
 	g, ctx := errgroup.WithContext(ctx)
 	var mux sync.Mutex
 	for i := range certs {
 		cert := certs[i]
 		g.Go(func() error {
-			validCertAliases, err := a.findValidAliasesAgainstCert(ctx, aliases, cert)
+			validCertAliases, domainsperCert, err := a.findValidAliasesAndAllowedDomainsAgainstCert(ctx, aliases, cert)
```

Comment:
```suggestion
			validCertAliases, sans, err := a.findValidAliasesAndAllowedDomainsAgainstCert(ctx, aliases, cert)
```
The `perCert` part is probably not needed if all the SANS that get returned are for the same `cert`.